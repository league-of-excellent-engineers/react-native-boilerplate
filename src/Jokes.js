import React from 'react';
import { Provider } from 'react-redux';
import {
    createStore,
    applyMiddleware,
    compose,
    combineReducers,
} from 'redux';
import { AppNavigation } from './screens/AppNavigation';
import { jokesApi } from './middleware';
import {
    jokesReducer,
    navigationReducer,
} from './reducers';


const Reducer = combineReducers({
    jokes: jokesReducer,
    navigation: navigationReducer,
});

const Store = createStore(
    Reducer,
    {},
    compose(applyMiddleware(jokesApi))
);

const Jokes =
    () => (
        <Provider store={Store}>
            <AppNavigation />
        </Provider>
    );

export { Jokes };
