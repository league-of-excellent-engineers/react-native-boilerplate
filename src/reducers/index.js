export { jokesReducer } from './jokes';
export { navigationReducer } from './navigation';
