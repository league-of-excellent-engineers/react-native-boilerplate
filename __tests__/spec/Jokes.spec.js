import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

// Import test object.
import { Jokes } from '../../src/Jokes';

describe('<Jokes>', () => {
    it('should render', () => {
        const wrapper = shallow(<Jokes />);
        expect(wrapper).toMatchSnapshot();
    });
});
