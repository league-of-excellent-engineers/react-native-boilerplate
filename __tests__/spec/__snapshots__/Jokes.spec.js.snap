// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Jokes> should render 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(NavigationComponent) />
</Provider>
`;
